src/app/views/sidebar/Sidebar.tsx

```diff
@@ -6,10 +6,25 @@ import { translateMessage } from '../../utils/translate-messages';
 import History from './history/History';
 import { ResourceExplorer } from './resource-explorer';
 import SampleQueries from './sample-queries/SampleQueries';
-export const Sidebar = () => {
+
+interface ISidebar {
+  currentTab: string;
+  setSidebarTabSelection: Function;
+}
+export const Sidebar = (props: ISidebar) => {
+
+  const onPivotItemClick = (item?: PivotItem) => {
+    if (!item) { return; }
+    const key = item.props.itemKey;
+    if (key) {
+      props.setSidebarTabSelection(key);
```

Comment:
*question:* Why can't the sidebar store this information. It looks like it's the only component that should care about which pivot item was selected

Reply:
The Sidebar is re-rendered a fresh each time it is collapsed. This means that all local state is 're-set'. This is why I lifted the state to the App component